### ISSUE TITLE
Add new dialog UI component

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#8475]
This brand new component is a wrapper around [a11y-dialog](https://github.com/KittyGiraudel/a11y-dialog). This ticket is only about the UI implementation of the generic component, not individual modals. [Design link](https://www.figma.com/file/h67EsVXdWsfu38WGGxWfpi/Wagtail-Design-System?node-id=2098%3A32512)

Reference screenshot with two variants:

![Image](https://user-images.githubusercontent.com/877585/166565253-56c101ec-43e8-4218-8c15-d2b8df8b755a.png)

## Expected components:

- Dialog
- Backdrop
- Close button
- Header / message (info / positive / warning / critical
- Heading (w/ icon off to the side)
- Content
- Footer

## Implementation notes

Tailwind utility classes for dialog "chrome", except buttons.

```jinja
{% dialog title="Schedule publishing" icon="calendar" message_status="info" message_heading="You don’t have permission to publish this page." message_description="Publishing will be scheduled once the publishing workflow has completed."  %}
  <p>Choose when this page should be published and unpublished</p>

   <label>(pseudo-code) Go-live date</label>
   […]
    <label>(pseudo-code) Expiry date</label>
   […]

    <button type="button" class="button-nice-padding">Schedule publishing</button>
    <button type="button" class="button-nice-padding-cancel" data-cancel-maybe-does-this-make-sense>Cancel</button>
{% enddialog %}
```

## Tasks

- Create new templatetags: `{% dialog %}`, `{% enddialog %}`, with their respective templates. Similarly to https://github.com/wagtail/wagtail/blob/e4d563e49287587257b054617dd9412d76e341df/wagtail/admin/templatetags/wagtailadmin_tags.py#L680-L698 but with two tags for open and close.
- Add to styleguide page
- Set up a11y-dialog with appropriate configuration
- Layout: max width, max height (based on viewport)
- Modal contents must be vertically scrollable if overflowing screen
- Flexible height for message
- Close button has 44px x 44px area
- Mobile support: mockup TBD
- 420px width on desktop, width TBD on smaller viewports

## Acceptance criteria

- [ ] Matches design 
- [ ] Implementation follows relevant ARIA practices
- [ ] Mobile support

## Testing / review plan

- UI testing in German by @stevedya 
- Mobile testing by @stevedya 
- z-index testing with sidebar, header, datetimepicker, Draftail, StreamField, side panel, floating footer actions @stevedya
- Accessibility tests by @thibaudcolas
- Code review by @thibaudcolas

**Comment by thibaudcolas** [comment]
Merged in #8513.
